### PR TITLE
Test and Doc test cleanup

### PR DESCRIPTION
**Pull request by cppwfs**
* Updated TaskExecutionService tests to replaced deprecated code.
* Updated TaskLogsDocumentation to use Awaitility instead of a thread.sleep.
* Updated StreamDeploymentDocumentation to remove unnecessary prints that cluttered the log.
* Updated StreamDeploymentDocumentation to remove unnecessary thread.sleeps.

### CODE REVIEW

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java [hunk 1]
```diff
@@ -163,13 +161,11 @@ public void streamUpdate() throws Exception {
 				post("/streams/deployments/{timelog1}", "timelog1")
 						.contentType(MediaType.APPLICATION_JSON)
 						.content(json))
-				.andDo(print())
 				.andExpect(status().isCreated())
 				.andDo(this.documentationHandler.document(
 						pathParameters(parameterWithName("timelog1")
 								.description("The name of an existing stream definition (required)"))
 				));
-		Thread.sleep(30000);
```
Comment (onobc) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java:172]: Hmmm. Why was this in here before?

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java [hunk 2]
```diff
@@ -202,13 +196,11 @@ public void rollback() throws Exception {
 		this.mockMvc.perform(
 				post("/streams/deployments/rollback/{name}/{version}", "timelog1", 1)
 						.contentType(MediaType.APPLICATION_JSON))
-				.andDo(print())
 				.andExpect(status().isCreated())
 				.andDo(this.documentationHandler.document(
 						pathParameters(parameterWithName("name")
 										.description("The name of an existing stream definition (required)"),
 								parameterWithName("version").description("The version to rollback to"))));
-		Thread.sleep(30000);
```
Comment (onobc) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java:211]: Ditto https://github.com/spring-cloud/spring-cloud-dataflow/pull/5349/files#r1197992799

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionServiceTests.java [hunk 3]
```diff
@@ -95,13 +95,10 @@
...
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionServiceTests.java:104]: I am not sure if this works w/ Junit4 test or not. I think so...

That being said, the other thing I have been doing when I go through tests is convert them over to Junit5. This one is still on Junit4.

**Review by onobc — APPROVED** [review]
@cppwfs once you get this to where you want it, go ahead and merge. Even w/ my comments it is LGTM.

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java (hunk 2 above)
Comment (cppwfs) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/StreamDeploymentsDocumentation.java:211]: Not sure why the thread.sleeps were necessary.  The skipper calls were mocked so, there should be no delay in awaiting for a response.   And on tests on CI and local there was no impact for removing these thread.sleep(s).

**Pull request merged**